plugins/pulp_rpm/plugins/importers/yum/repomd/metadata.py

```diff
@@ -333,6 +333,9 @@ def add_repodata(self, model):
             try:
                 db_file = gdbm.open(self.dbs[filename], 'r')
                 raw_xml = db_file[db_key]
+            except KeyError:
+                # Not found  # TODO: Make sure this is okay
```

Comment:
Every rpm should have an entry in "fileinfo" and "other", and those entries get written to this db_file. If that is not the case, we're in big trouble, and the publish would likely fail.